modules/afamqp/afamqp.c

```diff
@@ -654,6 +679,31 @@ afamqp_worker_insert(LogThreadedDestDriver *s, LogMessage *msg)
   return LTR_SUCCESS;
 }
 
+static void
+_handle_heartbeat(void *cookie)
+{
+  AMQPDestDriver *self = (AMQPDestDriver *) cookie;
```

Comment:
This timer shares the thread with amqp worker, that pushes the messages from the queue. If that queue happens to be non empty, and no suspend comes from the source, it means this thread does not have the chance to run the timer.

(the following is only my assumption) that could cause the amqp to unable to send message, so causing every now and than amqp to stop producing, resulting having a time to execute the timer, which shall fail but wait for `time_reopen` time till continue.

Could amqp gracefully handle this internally ?

Reply:
According to https://www.rabbitmq.com/heartbeats.html

> Any traffic (e.g. protocol operations, published messages, acknowledgements) counts for a valid heartbeat. Clients may choose to send heartbeat frames regardless of whether there was any other traffic on the connection but some only do it when necessary.

This means it does not matter if we send message or heartbeat, both will refresh the connection. We tried that we disabled this timer, but sent logs using `example-msg-generator` periodically, and the server did not terminate the connection. So this seems to work.

Reply:
Thanks for the clarification.